File: Palindrome_Number/palindrome_number.py
# ID:
# Tantangan 1: Palindrome Number
# Link: https://leetcode.com/problems/palindrome-number/
# Difficulty: Easy
# Tanggal: 2021-08-26
# ==================================
# Apa itu Palindrome Number? Palindrome Number adalah suatu bilangan yang jika dibalik, bilangan tersebut tetap sama.
# Contoh:
# 121 adalah Palindrome Number
# -121 bukan Palindrome Number
# 10 bukan Palindrome Number
# -101 bukan Palindrome Number

# ==================================
# Cara penyelesaian:
# Diberikan suatu bilangan bulat x, tentukan apakah x adalah Palindrome Number.
# ==================================


def is_palindrome(x: int) -> bool:
    if(x < 0):
        return False
    else:
        return x == int(str(x)[::-1])
# str(x)[::-1]
# str(x) adalah mengubah bilangan x menjadi string
# [::-1] adalah mengubah string tersebut menjadi string yang dibalik atau disebut reverse string
# int() adalah mengubah string tersebut menjadi integer


# Test case
print(is_palindrome(1221))  # True 

File: Palindrome_Number/palindrome_number.php
<?php
/*
    NOTE:
    - Palindrome number adalah bilangan yang jika dibalik akan tetap sama.
 */

//  Jawabannya
function is_palindrome($x)
{
    $x = (string)$x; // mengubah angka menjadi string
    $len = strlen($x); // menghitung panjang string

    // melakukan perulangan untuk mengecek apakah angka tersebut palindrome atau tidak
    // [1,2,2,1], Panjang array 4
    // [0,1,2,3]
    for ($i = 0; $i < $len; $i++) {
        if ($x[$i] != $x[$len - $i - 1]) {
            return false;
        }
    }

    return true;
}


// Test case
echo is_palindrome(121) ? "true" : "false"; // true
echo "\n";
echo is_palindrome(122) ? "true" : "false"; // true
